pkg/oauth/api/v1/types.go

```diff
@@ -79,8 +79,23 @@ type OAuthClient struct {
 
 	// RedirectURIs is the valid redirection URIs associated with a client
 	RedirectURIs []string `json:"redirectURIs,omitempty" description:"valid redirection URIs associated with a client"`
+
+	// GrantStrategy determines how to handle grants for this client. If no method is provided, the
+	// cluster default grant handling method will be used
+	GrantStrategy GrantHandlerType `json:"grantStrategy,omitempty"`
```

Comment:
Add a longer description here, including what each strategy does (so it shows up in swagger).